File: app/src/main/java/dev/benedek/syncthingandroid/activities/ThemedAppCompatActivity.java
package dev.benedek.syncthingandroid.activities;

import android.content.SharedPreferences;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

import dev.benedek.syncthingandroid.SyncthingApp;
import dev.benedek.syncthingandroid.service.Constants;

import javax.inject.Inject;

/**
 * Provides a themed instance of AppCompatActivity.
 */
public class ThemedAppCompatActivity extends AppCompatActivity {

    private static final String FOLLOW_SYSTEM = "-1";

    @Inject
    SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        ((SyncthingApp) getApplication()).component().inject(this);
        // Load theme.
        //For api level below 28, Follow system fall backs to light mode
        int prefAppTheme = Integer.parseInt(sharedPreferences.getString(Constants.PREF_APP_THEME, FOLLOW_SYSTEM));
        AppCompatDelegate.setDefaultNightMode(prefAppTheme);
        super.onCreate(savedInstanceState);
    }

}

File: app/src/main/kotlin/dev/benedek/syncthingandroid/activities/ThemedAppCompatActivity.kt
package dev.benedek.syncthingandroid.activities

import android.content.SharedPreferences
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.preference.PreferenceManager

/**
 * Provides a themed instance of AppCompatActivity.
 */
open class ThemedAppCompatActivity : AppCompatActivity() {

    val sharedPreferences: SharedPreferences by lazy {
        PreferenceManager.getDefaultSharedPreferences(this)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        // Load theme.
        //For api level below 28, Follow system fall backs to light mode
        val prefAppTheme = sharedPreferences.getString(
            dev.benedek.syncthingandroid.service.Constants.PREF_APP_THEME,
            FOLLOW_SYSTEM
        )!!.toInt()
        AppCompatDelegate.setDefaultNightMode(prefAppTheme)
        super.onCreate(savedInstanceState)
    }

    companion object {
        private const val FOLLOW_SYSTEM = "-1"
    }
}
